### PR TITLE
Ignore nonexistent globs

### PR DESCRIPTION
**Pull request by jszumski**
Prunes podspec patterns that don't match anything on disk.  This is needed for Bazel builds with  `--incompatible_disallow_empty_glob` enabled.  Invalid patterns typically show up in 2 ways:
- mistakes by the podspec author.  Examples from [Adyen](https://github.com/Adyen/adyen-ios/issues/997), [AardvarkCrashReport](https://github.com/cashapp/AardvarkCrashReport/issues/13), and [Aardvark](https://github.com/square/Aardvark/issues/133).
- intentional `**/*` patterns that are generously expanded into `**/*.{ext}` for each supported extension. Example from [GoogleDataTransport](https://github.com/google/GoogleDataTransport/blob/72f385de06190c832c85f8707f062237417749d2/GoogleDataTransport.podspec#L33):

    ```
    s.source_files = ['GoogleDataTransport/GDTCORLibrary/**/*',
                      'GoogleDataTransport/GDTCCTLibrary/**/*']
    ```
    turns into:
    ```
    name = "GoogleDataTransport",
    srcs = glob([
        "GoogleDataTransport/GDTCORLibrary/**/*.h",
        "GoogleDataTransport/GDTCORLibrary/**/*.hh",
        "GoogleDataTransport/GDTCORLibrary/**/*.m",
        "GoogleDataTransport/GDTCORLibrary/**/*.mm",
        "GoogleDataTransport/GDTCORLibrary/**/*.swift",
        "GoogleDataTransport/GDTCORLibrary/**/*.c",
        "GoogleDataTransport/GDTCORLibrary/**/*.cc",
        "GoogleDataTransport/GDTCORLibrary/**/*.cpp",
        "GoogleDataTransport/GDTCCTLibrary/**/*.h",
        "GoogleDataTransport/GDTCCTLibrary/**/*.hh",
        "GoogleDataTransport/GDTCCTLibrary/**/*.m",
        "GoogleDataTransport/GDTCCTLibrary/**/*.mm",
        "GoogleDataTransport/GDTCCTLibrary/**/*.swift",
        "GoogleDataTransport/GDTCCTLibrary/**/*.c",
        "GoogleDataTransport/GDTCCTLibrary/**/*.cc",
        "GoogleDataTransport/GDTCCTLibrary/**/*.cpp",
    ]),
   ```
   after this change it would be:
   ```
   name = "GoogleDataTransport",
   srcs = glob([
        "GoogleDataTransport/GDTCORLibrary/**/*.h",
        "GoogleDataTransport/GDTCORLibrary/**/*.m",
        "GoogleDataTransport/GDTCCTLibrary/**/*.h",
        "GoogleDataTransport/GDTCCTLibrary/**/*.m",
        "GoogleDataTransport/GDTCCTLibrary/**/*.c",
   ]),
    ```

The existing test for GlobExpansion already included paths that intentionally didn't exist on disk so I only needed to update its "after" fixture.

Relies on https://github.com/bazel-ios/cocoapods-bazel/pull/79.

### CODE REVIEW

File: lib/cocoapods/bazel/target.rb [hunk 1]
```diff
@@ -532,24 +532,28 @@ def expand_glob(glob, extensions: nil, expand_directories: false)
...
-              File.join(glob, '**', "*.#{ext}")
-            end
+              combined = File.join(glob, '**', "*.#{ext}")
+              combined if Dir.glob(File.join(@package_dir, combined)).any?
+            end.compact
           end
         elsif expand_directories
           if glob.end_with?('/**/*')
-            [glob]
+            Dir.glob(File.join(@package_dir, glob)).any? ? [glob] : []
```
Comment (thiagohmcruz) [inline lib/cocoapods/bazel/target.rb:547]: Can we extract this pattern to a small helper function since it's being used in diff places?

**Review by thiagohmcruz — COMMENTED** [review]
#79 just landed to this is probably good to rebase

File: lib/cocoapods/bazel/target.rb (hunk 1 above)
Comment (jszumski) [reply lib/cocoapods/bazel/target.rb]: Updated!

**Review by thiagohmcruz — APPROVED** [review]
Thx for the PR!

**Pull request merged**